be/src/exprs/agg/window_funnel.h

```diff
@@ -112,35 +115,231 @@ struct WindowFunnelState {
         }
     }
 
-    int32_t get_event_level() const {
+    int32_t get_event_level(Column* to) const {
         if (!sorted) {
             sort();
         }
 
-        std::vector<typename TimeType::type> events_timestamp(events_size, -1);
-        for (size_t i = 0; i < events_list.size(); i++) {
-            typename TimeType::type timestamp = (events_list)[i].first;
-            uint8_t event_idx = (events_list)[i].second;
+        if (!mode) {
+            std::vector<typename TimeType::type> events_timestamp(events_size, -1);
+            for (size_t i = 0; i < events_list.size(); i++) {
+                typename TimeType::type timestamp = (events_list)[i].first;
```

Comment:
why we use (events_list)

Reply:
> why we use (events_list)

Your means we should use other data struct(other than std::vector)?